### PR TITLE
Add metrics service gRPC gateway. Upgrade deps.

### LINKED ISSUE #607: all: update opencensus-proto to latest version and add grpc-gateway Metrics release is made
This issue is a placeholder to update to the opencensus-proto version that features Metrics for the grpc-gateway as per PR https://github.com/census-instrumentation/opencensus-proto/pull/205

and then in the source code, we can add this

```diff
diff --git a/receiver/opencensusreceiver/opencensus.go b/receiver/opencensusreceiver/opencensus.go
index 1db22ab..abeba11 100644
--- a/receiver/opencensusreceiver/opencensus.go
+++ b/receiver/opencensusreceiver/opencensus.go
@@ -259,6 +259,11 @@ func (ocr *Receiver) startServer() error {
 				errChan <- err
 				return
 			}
+			err = agentmetricspb.RegisterMetricsServiceHandlerFromEndpoint(c, ocr.gatewayMux, endpoint, opts)
+			if err != nil {
+				errChan <- err
+				return
+			}
 
 			// Start the gRPC and HTTP/JSON (grpc-gateway) servers on the same port.
 			m := cmux.New(ocr.ln)
```

and then finally the grpc-gateway will be able to receive metrics in JSON form.

### PR DESCRIPTION
**Pull request by songy23**
Fixes #607

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,17 +3,17 @@ module github.com/census-instrumentation/opencensus-service
 require (
 	contrib.go.opencensus.io/exporter/aws v0.0.0-20181029163544-2befc13012d0
 	contrib.go.opencensus.io/exporter/jaeger v0.1.1-0.20190430175949-e8b55949d948
-	contrib.go.opencensus.io/exporter/ocagent v0.5.0
+	contrib.go.opencensus.io/exporter/ocagent v0.5.1-0.20190621222907-f9129169651c // TODO: pin a released version
```
Comment (pjanotti) [inline go.mod]: just to confirm: there aren't changes to ocagent and stackdriver this is just to match the same proto version, correct?

Comment (songy23) [reply go.mod]: There're also some small bug fixes in both of them. You reminded me to upgrade the proto versions :) Just sent https://github.com/census-ecosystem/opencensus-go-exporter-ocagent/pull/74.

Comment (odeke-em) [reply go.mod]: Approved :)

**Comment by songy23** [comment]
Fixed dependency version and added a simple test.

**Pull request merged**